### PR TITLE
Convert solver to work in a 2D domain.

### LINKED ISSUE #4: Adapt the existing solver to work in a 2D domain.
Adapt the existing solver to work in a 2D domain.

### PR DESCRIPTION
**Pull request by oparry-ukaea**
Changes to Nektar++ equation sys and particle sys to make the solver work in 2D, plus a 2D example.

Resolves #4

### CODE REVIEW

File: examples/implicit2D/hw.xml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            <P> d22      = 0.0 </P> 
+            <!-- HW params -->
+            <P> HW_alpha = 2.0 </P>
+            <P> HW_kappa = 1.0 </P> 
+            <!-- Scaling factor for ICs -->
+            <P> s        = 2.0 </P>
+            <!-- No particles when num_particles_total = 0-->
+            <P> num_particles_total = 2000 </P>
+            <!-- Mass of each particle -->
+            <P> particle_mass = 0.5 </P>
```
Comment (will-saunders-ukaea) [inline examples/implicit2D/hw.xml]: We should make particle_mass and particle_charge 1 again I think - sorry I think I changed these and forgot to put them back..

Comment (oparry-ukaea) [reply examples/implicit2D/hw.xml]: Thanks, done

**Comment by oparry-ukaea** [comment]
I think this can be merged now.

@will-saunders-ukaea - could you make sure you can build it and have a quick scan through the changes?
It requires NESO commit 3279471308fefd934b95694d36cdcd42ec76bc97 from [this](https://github.com/ExCALIBUR-NEPTUNE/NESO/tree/tmp/hw-impurities-transport) temporary branch at the moment.

Once we've merged this, my short term plan is to
1. Merge ExCALIBUR-NEPTUNE/NESO#228
2. Make another NESO PR to update to the required nektar version
3. Write some build instructions for this repo (#7) which will amount to "clone NESO/main; build and activate spack env; build this app"

File: src/ParticleSystems/ParticleSystem.hpp [hunk 2]
```diff
@@ -94,17 +111,24 @@ class ParticleSystem : public PartSysBase {
...
+            vtmp += d(rng_phasespace);
+          }
+          initial_distribution[Sym<REAL>("VELOCITY")][px][dimx] = vtmp;
         }
         initial_distribution[Sym<REAL>("Q")][px][0] = particle_charge;
         initial_distribution[Sym<REAL>("M")][px][0] = particle_mass;
         initial_distribution[Sym<REAL>("B")][px][0] = Bx;
         initial_distribution[Sym<REAL>("B")][px][1] = By;
-        initial_distribution[Sym<REAL>("B")][px][2] = Bz;
+        initial_distribution[Sym<REAL>("B")][px][2] = Bz; ////
```
Comment (will-saunders-ukaea) [inline src/ParticleSystems/ParticleSystem.hpp]: Maybe stray ////?

**Comment by will-saunders-ukaea** [comment]
Updated my repos and have the implicit2D example running.

**Comment by will-saunders-ukaea** [comment]
Ran the implicit2D example and the particle output looks like what we've seen before.

File: src/ParticleSystems/ParticleSystem.hpp (hunk 2 above)
Comment (oparry-ukaea) [reply src/ParticleSystems/ParticleSystem.hpp]: Removed

**Comment by oparry-ukaea** [comment]
Happy to merge?

**Pull request merged**